### ISSUE TITLE
Translations for country and cache types

### ISSUE THREAD

**Issue by harrieklomp** [geokrety/geokrety-website#32]
Could it be possible to add translations for countries and cache types?
![selectie_156](https://user-images.githubusercontent.com/1060518/38429724-81d6691a-39bf-11e8-9d21-f82e97e817a7.png)

**Comment by kumy** [comment]
Hum, nice catch. Those comes directly from the database. The database is synchronized from different opencaching sites. First step would be to synchronize those with English terms instead of "native opencaching site language"? Then we can think of a way to translate. It'll not be that easy. @harrieklomp please don't be frustrated if we don't fix this using the current website. @harrieklomp, @boly38 If you agree, I would close it as "Won't do - in current site version".

**Comment by boly38** [comment]
Mixed workaround would be to create fieldset label 'opencaching result

**Comment by kumy** [comment]
I'm sorry, I don't understand your idea. May you develop a bit more please.

On April 6, 2018 9:12:23 PM GMT+02:00, Boly38 <notifications@github.com> wrote:
>Mixed workaround would be to create fieldset label 'opencaching result
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/geokrety/geokrety-website/issues/32#issuecomment-379349865

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by kumy** [comment]
@boly38 would you open a pull request / post a picture of the result?

**Comment by boly38** [comment]
yep (next week) 🙃

**Comment by boly38** [comment]
My personal draft / note:
* import waypoint [oc](https://github.com/geokrety/geokrety-scripts/tree/master/waypointy/oc)

**Comment by boly38** [comment]
Hi @kumy . 
I need your opinion.

about this usecase: **waypoint search gives one result**, example:
![image](https://user-images.githubusercontent.com/3100576/39124616-5904a368-46fc-11e8-8fb2-14f8ccfd6875.png)


In this case, `szukaj-ajax.php` append 2 informations

Exemple: 
* $typ == "Multicache"
* $kraj == "Österreich"

What do you think about using some kind of dynamic translation for theses 2 fields values ?
we could also mount a translation from this two codes :

     _('WPTYP_'.$typ)_
     _('WPKRAJ_'.$kraj)_

In our dictionary we will have some entries starting with prefix "WPTYP_" or "WPKRAJ_" to make frontier between waypoint data related entries and geokrety website legacy entries.

Seems the distinct values for this 2 fields are not so big: 10 cache type (Typ) and 35 country (kraj)
![distincttyp](https://user-images.githubusercontent.com/3100576/39124666-83597cf6-46fc-11e8-90dc-c127d4b4d740.JPG)
![distinctkraj](https://user-images.githubusercontent.com/3100576/39124668-886366a8-46fc-11e8-9497-fd2b0b5ac84c.JPG)


This result in adding 45 entry to crowdin.

What do you think ?

**Comment by kumy** [comment]
Yep, go for it. Don't forget to add new strings "in English" and probably in a dedicated file as current strings are extracted from source code and not database… A `yaml` file would be perfect for this,

**Comment by boly38** [comment]
>probably in a dedicated file as current strings are extracted from source code and not database… A yaml file would be perfect for this,
not understand that point, sorry

**Comment by kumy** [comment]
The strings uploaded to crowdin are - until now - extracted from source code using `tsmarty2c` and `xgettext`. See https://github.com/geokrety/geokrety-website/blob/8367a4dac3e4958eed41c1633660859dcaedaca9/scripts/export-translations-github.sh#L27-L29

If I'm not mistaken, you want to translate some strings that come from the database (table `gk-waypointy`). The point I raised was to solve this. "How to inform crowdin about those strings coming from the database?".

So, I suggest to extract/copy those 45 strings out of the database as a static file, as `yaml` for example. And use a tool to convert `.yml` file to `.po` file, like [yaml2po](https://github.com/unho/yaml2po). Then merge all our 3 files into one and finally upload it to crowdin. All this should happen in the script cited above L27-29.

Of course, the strings in the database need to be translated into English first, and also be in English in the `.yml` file.

**Comment by boly38** [comment]
ok thanks for your lights. I will see that and follow this way 🤸

**Comment by boly38** [comment]
hey, finally we have got 

**110 countries**

See https://github.com/geokrety/geokrety-website/blob/754b82a0023f79c1098dfd389a6ca44adb43792d/docker/mariadb/gk-waypointy-country.sql#L12-L121

**19 cache types**

See https://github.com/geokrety/geokrety-website/blob/754b82a0023f79c1098dfd389a6ca44adb43792d/docker/mariadb/gk-waypointy-type.sql#L12-L30

You could put an eye to see any error.
Next step: crowdin..

**Comment by boly38** [comment]
So I think we're no so far, 
https://github.com/geokrety/geokrety-website/blob/perso/boly/website/templates/waypointy-translations.html

is generated by

https://github.com/geokrety/geokrety-scripts/blob/feature/waypointy-translations/waypointy/oc/waypointy-translations.php

Next time will prepare pull request on both side
If you see huge error, please let me know
